app/scripts/modules/core/src/search/widgets/Filter.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import autoBindMethods from 'class-autobind-decorator';
+
+import { IFilterType } from './SearchFilterTypeRegistry';
+
+export interface IFilterProps {
+  filterType: IFilterType;
+  isActive: boolean;
+  onKeyUp(event: React.KeyboardEvent<HTMLElement>): void;
+}
+
+@autoBindMethods
+export class Filter extends React.Component<IFilterProps> {
+  constructor(props: IFilterProps) {
+    super(props);
+  }
+
+  private handleKeyUp(event: React.KeyboardEvent<HTMLElement>): void {
+    this.props.onKeyUp(event);
+  }
+
+  public render(): React.ReactElement<Filter> {
+
+    const { isActive } = this.props;
+    const className = classNames({
+      'filter': true,
+      'filter--focus': isActive,
+      'filter--blur': !isActive
+    });
+
+    const { modifier, text } = this.props.filterType;
+    return (
+      <div
+        role="option"
+        className={className}
+        onKeyUp={this.handleKeyUp}
+      >
+        <div className="filter__text">{text}</div>
+        <div className="filter__modifier">[{modifier.toLocaleUpperCase()}:]</div>
```

Comment:
I didn't realize that was coming out of BEM.

The BEM entry was more aspirational than anything, as we really haven't followed it at all since I put that in the conventions document a year ago.

The rule of thumb is to follow the coding conventions, but if the actual code conflicts with the conventions, we should revise the conventions so they're aligned with how we're writing code, or change how we write code.

I like the idea of BEM and would be happy to start using it more frequently, like here, but will defer to others.